testing-overhead-jmh/build.gradle.kts

```diff
@@ -0,0 +1,54 @@
+import net.ltgt.gradle.errorprone.errorprone
+
+plugins {
+  id("otel.java-conventions")
+  id("otel.jmh-conventions")
+}
+
+val agentJar by configurations.creating {
+  isCanBeConsumed = false
+  isCanBeResolved = true
+}
+
+dependencies {
+  jmhImplementation("org.springframework.boot:spring-boot-starter-web:2.5.2")
+  jmhImplementation(project(":testing-common")) {
+    exclude("ch.qos.logback")
+  }
+
+  // this only exists to make Intellij happy since it doesn't (currently at least) understand our
+  // inclusion of this artifact inside of :testing-common
+  jmhCompileOnly(project(path = ":testing:armeria-shaded-for-testing", configuration = "shadow"))
+
+  agentJar(project(path = ":javaagent", configuration = "shadow")) {
+    isTransitive = false
+  }
+}
+
+tasks {
+  val copyAgent by registering(Copy::class) {
+    into(file("$buildDir/agent"))
+    from(configurations.named("agentJar"))
+    rename { file: String ->
+      file.replace("-$version", "")
+    }
+
+    dependsOn(":javaagent:shadowJar")
+  }
```

Comment:
gradle experts: let me know if there is a more preferred way to get the opentelemetry-javaagent-all into a location that the jmh benchmark can depend on